README.md

```diff
@@ -14,12 +14,23 @@ If there is anything you are looking for that is not covered here you should ref
 
 ### File structure ###
 
-You should not define multiple public/internal types (ie class, struct, enum) in the same file; each type should have its own file.
+You should not define multiple public/internal types (ie class, struct, enum) in the same file; each type should have its own file. If an enum or protocol is related to the class but has to be known outside define it inside the class file.
```

Comment:
I understand your point but that's against what we were used to do in Objective-C by having enum in the header file related to the class where it's used. Are we considering enums as separate types not related to a class or more a helper type that should live with the feature it's related to?